lib/identity_config.rb

```diff
@@ -0,0 +1,80 @@
+class IdentityConfig
+  class << self
+    attr_reader :store
+  end
+
+  CONVERTERS = {
+    string: proc { |value| value.to_s },
+    comma_separated_string_list: proc do |value|
+      value.split(',')
+    end,
+    integer: proc do |value|
+      Integer(value)
+    end,
+    json: proc do |value, options: {}|
+      JSON.parse(value, symbolize_names: options[:symbolize_names])
+    end,
+    boolean: proc do |value|
+      case value
+      when 'true', true
+        true
+      when 'false', false
+        false
+      else
+        raise 'invalid boolean value'
+      end
+    end,
+  }
+
+  def initialize(read_env)
+    @read_env = read_env
+    @written_env = {}
+  end
+
+  def add(key, type: :string, is_sensitive: false, options: {})
+    value = @read_env[key]
+    raise "#{key} is required but is not present" if value.nil?
+    converted_value = CONVERTERS.fetch(type).call(value, options: options)
+    raise "#{key} is required but is not present" if converted_value.nil?
+
+    @written_env[key] = converted_value
+    @written_env
+  end
+
+  def self.build_store(config_map)
+    config = IdentityConfig.new(config_map)
+    config.add(:admin_email, type: :string)
+    config.add(:asset_host, type: :string)
+    config.add(:assets_version, type: :string)
+    config.add(:auto_account_creation_tlds, type: :string)
+    config.add(:aws_region, type: :string)
+    config.add(:aws_logo_bucket, type: :string)
+    config.add(:db_pool, type: :integer)
+    config.add(:dp_reaping_frequency, type: :integer)
+    config.add(:certificate_expiration_warning_period, type: :integer)
+    config.add(:dashboard_api_token, type: :string)
+    config.add(:email_recipients, type: :string)
+    config.add(:idp_sp_url, type: :string)
+    config.add(:idp_url, type: :string)
+    config.add(:logo_upload_enabled, type: :boolean)
```

Comment:
I'd really like us to move this class into hostdata soon, and then make this long `build_store` more DSL-y and move it into `application.rb` or some other initializer

Reply:
Agreed, I was thinking through that while putting this together.